splice_machine/src/test/java/com/splicemachine/derby/utils/SpliceDateFunctionsIT.java

```diff
@@ -1232,4 +1232,15 @@ public void testInsertTimestampDuringDaylightSavingGap() throws Exception {
     //====================================================================================================
     // DB-2248, DB-2249: BI Tools and ODBC functions - end
     //====================================================================================================
+
+    //DB-6790
+    @Test
+    public void testToDateWithIncorrectFormatPattern() throws Exception {
+        String sqlText = "values to_date('2018-12','MM-DDD')";
+        try (ResultSet rs = methodWatcher.executeQuery(sqlText)) {
+            fail("Expected exception the syntax of the string representation of a datetime value is incorrect");
+        } catch (SQLDataException e) {
+            assertEquals("The syntax of the string representation of a datetime value is incorrect.", e.getLocalizedMessage());
```

Comment:
Please check the SQLState rather than the message (which might be locale dependant)